pallets/pips/src/lib.rs

```diff
@@ -1289,13 +1299,16 @@ impl<T: Trait> Module<T> {
     /// Unschedule PIP with given `id` if it's scheduled for execution.
     fn maybe_unschedule_pip(id: PipId, state: ProposalState) {
         if let ProposalState::Scheduled = state {
-            Self::remove_pip_from_schedule(<PipToSchedule<T>>::take(id).unwrap(), id);
+            Self::unschedule_pip(id);
         }
     }
 
     /// Remove the PIP with `id` from the `ExecutionSchedule` at `block_no`.
-    fn remove_pip_from_schedule(block_no: T::BlockNumber, id: PipId) {
-        <ExecutionSchedule<T>>::mutate(block_no, |ids| ids.retain(|i| *i != id));
+    fn unschedule_pip(id: PipId) {
+        <PipToSchedule<T>>::remove(id);
+        if let Err(_) = T::Scheduler::cancel_named(Self::pip_execution_name(id)) {
```

Comment:
Ignore this suggestion if you'd rather the current `if let Err(_) = ...` expression.
```suggestion
        if T::Scheduler::cancel_named(Self::pip_execution_name(id)).is_err() {
```

Reply:
Direct pattern matching saves us from an extra function call that is not optimised in WASM. So I think `if let Err(_) = ...` is marginally more performant.